### ISSUE TITLE
Function to assign a conversation to different contact after creation

### ISSUE THREAD

**Issue by devmato** [chatwoot/chatwoot#5115]
**Is your feature request related to a problem? Please describe.**
every time a new conversation is started trough the website inbox a new contact with a random name is generated. i couldn't find any possibility to reassign the new conversation to an existing contact wich has been previously imported from our CRM (by api)

The only way i could find is to use the "merge contact" function in the webapp, but the direction of the merge action cannot be changed so the existing contact will get deleted and the newly created random contact will be kept

**Describe the solution you'd like**
Is there any possibility to assign a conversation to a different existing contact after the conversation has been created over the website inbox or at least a way to change the direction of the merge function so that the existing contact will be kept and the newly created contact with a random name would be deleted?

**Additional context**
Our plan is to use Chatwood as additional possibility for our customers to contact us. After the conversation has been closed an self written software should import the conversation into our existing ticketing system by using the Chatwood API. For the customer identification we plan to use the identifier field of the Chatwood customer database. Our problem is that currently we cannot assign conversations to existing customers because there seems to be no way to do this either over the Chatwoot webapp or the API itself. 

Do you have a solution for this? Maybe i have understood something wrong and this requires just a different workflow to achieve the desired result?

**Comment by sojan-official** [comment]
have you explored using `setUser` method in our SDK .
 https://www.chatwoot.com/docs/product/channels/live-chat/sdk/setup/#set-the-user-in-the-widget

**Comment by devmato** [comment]
Yes i know the setUser  method over the Javascript SDK, but unfortunately this isn't usable for us, because our backend isn't accessible over the public internet and the website where the chat widget is embedded is just a regular static page.

As i have mentioned before chatwoot is just a secondery system for us and i have considered the following workflow:
- Customer starts a chat on our website and we ask via the pre chat form and a custom attribute for the customer number (identifier) or the customer name (for the case the customer doesn't know his customer number)
- After the conversation has beend started our backend gets called via the chatwoot webhook and tries to research the customer number in our database. In case the customer can be found in our database, the backend creates the customer in the chatwoot database via api and should be able to assign the conversation to the newly created contact (wich right now seems not the possible)
- As soon as our helpdesk worker sets the conversation to resolved our backend gets called again via webhook and tranfers the conversation to our ticketing system

Am i right with the assumption that right no it isn't possible to reassign a conversation to a different contact once the conversation has been created, or is there some hidden method for that?

**Comment by sojan-official** [comment]
@devmato conversations can't be moved between contacts. But maybe you can make this work using the merge contact. 

 > The only way I could find is to use the "merge contact" function in the web app, but the direction of the merge action cannot be changed, so the existing contact will get deleted, and the newly created random contact will be kept.
 
Merge contact functionality works with the current contact as the base contact. 
In your case, during step 2: After you find or create the appropriate contact in Chatwoot Database via API,  just merge the contact id with a random name into the contact you found over API?  This merge can also be done via API 


```
POST:   api/v1/accounts/1/actions/contact_merge
PARAMS: { 
                    base_contact_id: 3
                    mergee_contact_id: 1
                }

```

- base contact should be the id of the contact you created/found via API 
- mergee contact id can be the contact with the random name ..

**Comment by devmato** [comment]
i will try to resolve my problem by using the contact_merge API. thank you!

Anyway i think this two features would be very helpful for similar use cases. Are they planned to be added at sometime to Chatwoot?
- Choose direction when merging contacts via webui
- Assign conversations to different contact at least via api

**Comment by ChefAndy** [comment]
I'll tack on another vote for this functionality, but for the GUI also. While it doesn't seem as pertinent in a chat-first workflow, I've got a non-support email address people will email accidentally and with my current solution, I can just forward that email to the inbox address and have whoever picks it up just add the existing appropriate contact in there. No reason to confuse the client with any of the implementation details on our end, change any existing contact entries, or anything like that.
